src/gabarit_sup.erl

```diff
@@ -2,34 +2,36 @@
 %% @doc gabarit top level supervisor.
 %% @end
 %%%-------------------------------------------------------------------
-
 -module(gabarit_sup).
-
 -behaviour(supervisor).
-
 -export([start_link/0]).
-
 -export([init/1]).
-
 -define(SERVER, ?MODULE).
 
 start_link() ->
     supervisor:start_link({local, ?SERVER}, ?MODULE, []).
 
-%% sup_flags() = #{strategy => strategy(),         % optional
-%%                 intensity => non_neg_integer(), % optional
-%%                 period => pos_integer()}        % optional
-%% child_spec() = #{id => child_id(),       % mandatory
-%%                  start => mfargs(),      % mandatory
-%%                  restart => restart(),   % optional
-%%                  shutdown => shutdown(), % optional
-%%                  type => worker(),       % optional
-%%                  modules => modules()}   % optional
 init([]) ->
     SupFlags = #{strategy => one_for_all,
                  intensity => 0,
                  period => 1},
-    ChildSpecs = [],
-    {ok, {SupFlags, ChildSpecs}}.
+    %% TODO: which option is better for the application
+    StoreOpts = case application:get_env(gabarit, persistence, none) of
+                    none -> [];
+                    dets ->
+                        BaseDir = application:get_env(gabarit, dets_dir, "priv/gabarit_store"),
+                        [{persistence_module, gabarit_dets_persistence}, {persistence_dir, BaseDir}];
+                    mnesia ->
+                        [{persistence_module, gabarit_mnesia_persistence}]
+                end,
+
+    ChildSpecs = [
```

Comment:
same thing with children, if they are not dynamic, creating a function somewhere in the code to spawn them if necessary is always cool. Something like that:

```erlang
-spec gabari_store_spec() -> supervisor:child_spec().
gabarit_store_spec() ->
  #{
    ...
  }
```